src/router/simpleswap.ts

```diff
@@ -232,9 +234,24 @@ abstract class SimpleRouter implements IRouter<SimpleSwapParam> {
   ) {
     if (srcAmountWeth === BigInt('0') && destAmountWeth === BigInt('0')) return;
 
+    // Get dexKey from network
+    // It assumes each network has unique wrapped exchange
+    const wrappedExchangeKey = Weth.dexKeysWithNetwork.reduce(
+      (prev, current) => {
+        if (
+          prev === '' &&
+          current.networks.includes(this.dexAdapterService.network)
+        ) {
+          return current.key;
+        }
+        return prev;
+      },
+      '',
+    );
+
```

Comment:
this can be done in the constructor or outside such that we don't do this every-time we build tx

Reply:
Moved to constructor